### ISSUE TITLE
Dodging Archon projectiles in a Puffin isn't fun

### ISSUE THREAD

**Issue by Pointedstick** [endless-sky/endless-sky#7231]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Missions involving involve traveling to Nenia in a Puffin and dodging Archon fire to land on one of the planets are extremely frustrating. The fragile ship is instantly destroyed with one hit from the Archon's long-range weapon and dodging it is challenging, which means the player will die many times while trying to complete this mission and need to save-scum to complete it. This breaks the player's immersion because in effect it means the mission is a suicide mission and their captain would have to be a moron to take it on. Finally, the required travel time from Remnant space to Nenia makes save-scumming a long and tedious process, further reducing fun.

### Steps to Reproduce

Play one of the missions that involve traveling to Nenia in a Puffin and dodging Archon fire to land on one of the planets.

### Expected Behavior

The missing aren't an un-fun, time-consuming, immersion-breaking grind.

### Operating System

Fedora 36

### Game Source

Built from source

### Game Version

310c43399fd52a90fd97628346c296a8fac128c0

**Comment by Zitchas** [comment]
I'm not sure how we can improve this at this point: While there have been more than a few people complaining about it, I have also received quite a few comments from people describing how rewarding it is. 

As far as that goes, most of the complaints revolved around the old behavior where the Archon was sometimes right on top of the player when they took off, and as a result the player gets insta-killed before they can even react. This was fixed in https://github.com/endless-sky/endless-sky/pull/4977 and tweaked in https://github.com/endless-sky/endless-sky/pull/5321 . Since then, this issue is the first time I can recall hearing anyone (running a version of the game with these changes) complain about the puffin/archon missions.

From my own perspective, one reason why the "dodge the Archon shots" missions are so hard is the fact that dodging isn't a thing in ES. Throughout virtually the entire game one either tanks the shot, outruns the shot, or, in the case of missiles, has something to shoot the shot down. Dodging is never really a thing.

It is my hope that https://github.com/endless-sky/endless-sky/pull/6260 is merged people will start to develop more dodging abilities; and most importantly, people actually starting thinking about dodging as a valid way to avoid at least some kinds of incoming fire.

Specifically, the heavy rockets. These were the munitions that, in EV, trained me to dodge. In EV, for most of the early game when I was flying small to medium ships, getting hit by a heavy rocket (assuming it didn't kill me instantly) generally meant it was time to run because I was going to lose the battle. On the flip side, though, I eventually learned how to dodge them and handle the ships that had them accordingly. In ES, though, heavy rockets are significantly faster, harder to dodge, and the AI carries so many of them that baiting ships into using up their ammo is a much longer, harder, and often fruitless pursuit; so people get back into the habit of just tanking or shooting them with AM.

**Comment by Pointedstick** [comment]
Teaching the player to dodge by putting them in a situation where they die if they fail is kind of like teaching a kid to swim by throwing him in a lake, no? The best teaching method makes the material fun, not a dangerous, grueling grind.

In addition, I'm not sure the skills being "taught" here are transferable: the Puffin is small, highly maneuverable, and capable of coming to a complete stop on its own. And the projectiles being dodged are exceptionally fast with a large surface area. These characteristics do not apply to most ships the player will be flying or projectiles they'll be dodging. I don't have much difficulty dodging normal human missiles in a normal small ship, but I have extreme difficulty dodging the Archon's projectiles because it's a completely different skillset so I have to learn it from scratch, and every failure takes a while to recover from so I can try again.

I think it might help if you could take one hit and only the second or third one one killed you. Being so unforgiving that a single hit kills you also kills my enjoyment and has made me ragequit the game twice. Alternatively, maybe the projectiles' deadliness could be preserved if they either weren't as fast, or weren't as physically large.

**Comment by Pointedstick** [comment]
Some random observations today:

- Since yesterday, I attempted the "return the samples" mission 32 times before finally succeeding, which is in addition to the number of times I tried it yesterday that I didn't keep track of.
- Traveling from Arculus to Nenia in a Puffin takes 45 seconds, so the feedback loop is slow. Every failure necessitates 45 seconds before you can try again.
- Ion storms sometimes appear in Nenia, disabling your ship and making you unable to dodge the Archon's shots, while the Archon remains able to fire them with perfect accuracy. This happened in 3 (9.4%) of my attempts.
- The Archon's projectiles deal splash damage, so if one of them impacts an asteroid near you, you'll get killed even if you dodged it. This exact issue prevented me from completing the mission in 2 (6%) of my attempts when I got close and otherwise would have done it, in addition to all the times it happened when I was farther out.


In terms of strategies you can employ to try to learn your way out of the challenge:
- The Archon is fast, so trying to draw it away from the planet and then making a dash for it doesn't work.
- The Archon's projectile submunitions spread out with distance, making dodging harder than with any Human missile if you try to keep your distance from it.
- Zooming out to the max zoom level so you can see the projectiles you need to dodge for longer helps, but isn't applicable in any way to dodging Human missiles, and is also impaired anyway by the submunition spread.


The way I finally landed on Nasqueron actually didn't involve dodging the archon's projectiles; I first dodged two wile getting as close as possible, taking advantage of the fast that its close-range weapons inexplicably don't kill the Puffin instantly. So I tanked a few of its shots while stating just out of its range to bait it closer and then finally landed on the planet. It was the classic "too close for missiles, switching to guns" approach. However this in no 

So I don't think this mission teaches you missile dodging skills that are generally applicable, which means the annoyance it imposes on you is actually rather pointless.

**Comment by Pointedstick** [comment]
Recommendations:
- Remove the Antimatter Cannon's splash damage
- Reduce the Antimatter Cannon's projectile speed
- Reduce the Antimatter Cannon's submunition spread
- Reduce the Archon's speed
- Add an uninhabited planet you can land on in the Cardea system so you can load your game closer to Nenia after you fail and die

**Comment by Zitchas** [comment]
Honestly, I'm quite confused. I think I'm going to have to go back and do the missions, because the experience and behavior you are describing make it sound like something's broken.

**Comment by Pointedstick** [comment]
Either that, or it's not hard to you anymore because you already played it 1000 times while writing and modifying it. :)

**Comment by Zitchas** [comment]
For instance, the fact that you manage to get within range of its close-range turret at all is... Problematic. The Archon is *supposed* to have a repelling field that both keeps most ships out of range of its close-quarters turret; and b) kicks small ships that happen to be within that radius when you take off hard enough that the close quarter turret's shots don't connect (or only land a single hit - not enough to inflict lasting harm on the Puffin)

**Comment by Zitchas** [comment]
If that's not working, what *else* isn't working?

What sort of rate of fire are you seeing on the AMC? Is it a shot every couple seconds, or more like a stead stream?

**Comment by Pointedstick** [comment]
I think the repelling field worked after I took off. The moment the little Puffin took off from the planet, it was catapulted a million miles away without being destroyed. Which was actually convenient, since I *wanted* to be far away from the Archon!

> What sort of rate of fire are you seeing on the AMC?

It's one shot every couple seconds.

**Comment by Zitchas** [comment]
OK, good, that was the repelling field doing its job.

**Comment by Zitchas** [comment]
Originally implemented because, seeing as the Drak want to preserve life, it seemed likely they'd figure out a way to prevent lower tier species from suiciding against them. So, the keep away field, which doubles as the get-away-from-me-before-my-turrets-kill-you field.

**Comment by Pointedstick** [comment]
Honestly though, if the field had worked when I tried the "get in the Archon's face" approach, I'd probably still be struggling through my 60th attempt right now.

What is the intended way for the player to successfully land on one of these planets while the Archon is shooting at them?

**Comment by Zitchas** [comment]
Typically, the mechanics of the Archons weapons work such that if you charge the Archon, they keep moving back. So towards them while dodging side to side in sort of a W pattern is what has kept me alive for the past few years.

Just thinking about it, the recent change to the way force affects ships (removing the weighting that softens out the force towards 400) may have change how the Archon moves, which means that'll need to be tweaked

**Comment by Zitchas** [comment]
The Archon weighs in at over 400, so the recent change would have it being pushed slower by its recoil than it used to be. (conversely, the Puffin is <400, so it'll react to the repeller field *more* than it used to)

**Comment by Pointedstick** [comment]
> Typically, the mechanics of the Archon's weapons work such that if you charge the Archon, they keep moving back.

That's definitely not what I found.

It's also pretty counter-intuitive behavior that requires developer knowledge of the Archon's mechanics to understand that running towards it will make it go backwards. That's not really a logical thing for it to do (it's huge and isn't scared of you) so there's really no reason for the planet to want to try it because it doesn't make any sense. The most logical thing to do is to run the heck away from it while dodging its projectiles, and try to circle back to the planet, outrunning it there. But this doesn't work because of how fast it is. The only reason why I tried running toward it anyway was because I was getting desperate and running out of ideas.

**Comment by Zitchas** [comment]
For what it is worth, the initial "dodge the Archon" mission predates my involvement in ES. No developer knowledge there. :)

But yeah, slowing down the archon, increasing the pushback to keep it moving backwards, and adding in a pitstop somewhere in an adjacent system are all potential facets of the solution, I think.

**Comment by RisingLeaf** [comment]
I just want to mention this to get another view on this, but I got this on second try

**Comment by Hecter94** [comment]
Gave this a try myself since the experience that Pointed is having with this mission seemed very different from how I remember the mission going.
You can see my attempt here: (Drive Link since it's far too big to be attached directly)
https://drive.google.com/file/d/1QvfXm6PIIH_TsrV8yoOvhphBjZVprR7p/view?usp=sharing

This was my first attempt at this mission in several months, and while I was able to beat it on my third attempt, I noticed a couple of weird factors. 

For instance, at around the 40-second mark, you can see I take off from Nasqueron and get pushed back a significant distance due to hit force, which is what led to my first death. 
My second death is due to the Antimatter Cannon projectile hitting one of my escorts and blowing me up as well with the AoE damage. 
After some investigating, it seems what I'm getting pushed back by is the Augmented Drak Draining Field, and what I was killed by on my second attempt was the Augmented Antimatter Cannon which it's using since it is missing the `frugal` personality:
```
	npc
		uuid 64864212-df6d-4be2-849e-abf7aaae7575
		government "Drak (Hostile)"
		personality
			confusion 10
			heroic
			nemesis
			staying
			uninterested
		ship Archon
			name "Lifted Lorax"
			noun entity
```

Removing the Augmented Draining Field stopped that excessive pushback from occurring and removing the augmented antimatter cannon made the AoE much more forgiving. Those two changes made the mission much easier. 

In conclusion, I would personally say that the mission isn't overly difficult as it is now, however it is pretty frustrating with the excessive pushback and dying to AoE damage.
Adding the `frugal` personality to this Archon and preventing it from using it's augmented weapons would hopefully fix both of those issues.

**Comment by Pointedstick** [comment]
@Hecter94 thanks for posting that video. I see a major difference between your playthrough and mine. In yours, Nasqueron isn't between you and the Archon when you jump into the system. When I did the mission, the Archon was positioned directly between me and the planet.

Maybe this was just me getting screwed by the feature to randomly rotate how the system's stellar objects are arranged? The mission is much much harder if the Archon is between you and the planet.

**Comment by ravenshining** [comment]
What if instead of one-shot kill, it was one-shot *disabled*?  Together with an implementation of #4678 for the Puffin, and a `disables` personality on the Archon, you'd still have the dodge and try again challenge, but you'd get at least *one* second chance.  Maybe a couple chances, if the Archon is given a scanner, causing them to close the gap once you're disabled for a scan.  More, if you have escorts.

**Comment by Zitchas** [comment]
If #4678 was a thing, that could be a definite possibility. Although we'd need to really nerf the AMC to eliminate the ability to kill the puffin. (or buff the puffin). 

As far as that goes, how does the application of damage go? If a weapon has hull damage and disabled damage, if the hull damage value is enough to exceed 100% of the puffin's hull; but the disabled damage is, just for the sake of discussion, 1 point of damage. Will the weapon switch from doing hull damage to doing disabled damage the instant the ship flips over to being disabled? So in this case, the hit would take all of the puffin's hull right down to the disabled mark, and then do a single point over it? Or is the disabled damage simply invoked when the ship is already disabled when the shot impacts it?

**Comment by movingpictures** [comment]
I've never really had much of a problem with this mission. I have learned to leave everything parked, run the game on the non-accelerated setting and be patient. Dodging is not that hard.

**Comment by yafflehk** [comment]
Maybe have the Remnant help out by sending an escort to drop a couple of drones as a distraction? Or have some aspect of the Puffin's archaic shields (which are mentioned to the player, but not to any clear purpose), throw of the Archon projectiles, Plume can talk about that perhaps? The puffin is a leftover, according to lore, of the days when the Remnant would harvest the void sprites, give it countermeasures of some sort?

**Comment by Zitchas** [comment]
I believe this has now been fixed by #7239 , so I'm closing this. If we need to tweak it a bit more, though, we can definitely either re-open this or start a new issue highlighting whatever the current problems are.

**Comment by ThrawnCA** [comment]
> I've never really had much of a problem with this mission. I have learned to leave everything parked, run the game on the non-accelerated setting and be patient.

The fact that you need to park every other ship is itself an obstacle to fun, because it means you instantly fail any outstanding delivery/passenger missions.

> Dodging is not that hard.

Dodging a cannon that fires from further away than you can see, on a ship that will repel you *without the game explaining that that is a thing or how it works* so you end up going in a different direction from what you expected, to a destination that requires the landing command and therefore has to switch to autopilot at some point, in a situation where every retry requires you to start over from several systems away and make a long trek through multiple wormholes to try again?

A bit harder.

**Comment by movingpictures** [comment]
Disagree vehemently. Took a break from ES, now working through a nightly
build - early November, IIRC - where the mission to disable and board a
geocoris as part of the enhanced Hai timeline - that took me EIGHT
playthroughs to get right.

Archon doding? Two playthroughs, and only because in one, the Archon was on
top of Nasquerine on departure.  Take the puffin, dump any Korath hunting
missions, run on standard speed. Lure the archon away from the system core.
Watch the mini-display to observe when the bolts are coming, and dodge
accordingly.

**Comment by Zitchas** [comment]
> to a destination that requires the landing command, so you have to switch to autopilot at some point while still under fire, in a situation where every retry requires you to start over from several systems away and make a long trek through multiple wormholes to try again?

On the first part, I agree. I rather wish the landing autopilot could be disabled. I think it detects from the game.

In the second, as The Remnant mentions in those missions now, there is a gas giant in the last system before jumping to Nenia that he explicitly mentioned past Remnant have considered good luck to visit before going to Nenia. In practical terms, this gives you a save/reload point that is one hyperjimp away. (This is new as of 0.9.16)

**Comment by justinclift** [comment]
> If that's not working, what else isn't working?

The game not recognising when people have actually fulfilled the conditions its asking for in this quest is one. :frowning:

![2023 06 23-04 06-quest_refusing_to_complete_even_though_ive_done_exactly_what_its_asking_for_multiple_fucking_times](https://github.com/endless-sky/endless-sky/assets/406299/7121ca46-c540-470b-b9c6-2a7f03fc58ae)

At first I thought it might have been complaining as I visited them separately (eg exit the system after visiting the first tone, came back for the 2nd one later).

But nope.  I just went and did them both together in the same go, and it's *still* throwing this message.

Really hoping this isn't a game breaking bug.  I'd rather not have to edit save files, but if needed then I'll figure it out.

---

For the dodging Archon projectiles thing... while it wasn't great I didn't think it was too terrible.  Was reasonably straight forward to draw the thing away from the first planet and circle around it to get to them.

The bit where your escort ships launch from the planet alongside you afterwards - right near the Archon - was a bit more surprising, and resulted in their deaths (thus game reload time) in a few runs.

**Comment by Zitchas** [comment]
> The bit where your escort ships launch from the planet alongside you afterwards - right near the Archon - was a bit more surprising, and resulted in their deaths (thus game reload time) in a few runs.

This might be your problem here: The game often puts mission cargo onto escorts, and your escorts can't actually land on either of those gas giants. So technically, despite the fact that *you* visited the planets, the scanning equipment did not. 

I may have to add to that mission description something to the effect of recommending that you do not take escorts on this mission. 

Anyway, it's not guaranteed that this is the problem in this particular case; but if you could test it out by redoing the mission (or if you still have it active, just visiting the two planets again), except this time with all your escorts parked planetside on Aventine, it'd be much appreciated. If you can then complete the mission, that's what the problem was.

**Comment by warp-core** [comment]
> This might be your problem here: The game often puts mission cargo onto escorts, and your escorts can't actually land on either of those gas giants. So technically, despite the fact that you visited the planets, the scanning equipment did not.

Mission cargo doesn't need to be with you when visiting a stopover for it to count as having been visited.
I expect the mission cargo just wasn't at the destination.

**Comment by Zitchas** [comment]
Hmmm... That's problematic in its own right, then. Not much point in having required cargo for a mission if the mission will complete anyway with the cargo on an escort that can't go where the mission needs the cargo to go.

**Comment by warp-core** [comment]
Additionally, until https://github.com/endless-sky/endless-sky/commit/0447506d7b113b94b09d80fdf845d56452122f45, escorts would actually count as being landed on the gas giants if the flagship was (that's how they take off with the player.)

And, the save file seems to indicate that the stopovers have actually been visited.

**Comment by justinclift** [comment]
> So technically, despite the fact that you visited the planets, the scanning equipment did not.

If that's indeed the problem here, then something needs to tell the user.  Even something a dialog along the lines of:

> You dive to the depths of the gas giants... but with your scanning equipment on a different ship you're unable to record the results.

Hmmm, this is the save file in question, in case it helps: [Some Dude.txt](https://github.com/endless-sky/endless-sky/files/11839557/Some.Dude.txt)

**Comment by justinclift** [comment]
> if you still have it active, just visiting the two planets again[...], except this time with all your escorts parked planetside on Aventine, it'd be much appreciated.

No worries.  I'll give that a go now.

**Comment by justinclift** [comment]
Somehow managed to visit both planets without being killed, on my first attempt.  That's with *all* escorts parked.

Unfortunately, exactly the same dialog/behaviour upon returning:

![image](https://github.com/endless-sky/endless-sky/assets/406299/865735b6-9150-4b5f-8e78-3d54f0c2dbd7)

Saved the game right afterwards, which is this: 
[Some Dude~3026-09-23 - did mission but same error.txt](https://github.com/endless-sky/endless-sky/files/11839699/Some.Dude.3026-09-23.-.did.mission.but.same.error.txt)

**Comment by justinclift** [comment]
Note that the escorts were all **parked**, but not all on Aventine as they're spread throughout the map.  Guessing the parking thing was the important point, not their location.

**Comment by warp-core** [comment]
The mission can't complete because the cargo is on the Rano'erek named name "At Markarati" which is parked in the Segesta system.

**Comment by justinclift** [comment]
Ahhhh, thanks.  That makes sense as it's my flagship.  So switching to this runabout thing didn't automatically get the item the mission requires then.

I'll manually switch it now (if possible).  Pretty sure that should be automatic though. :smile:

**Comment by justinclift** [comment]
Hmmm, I can't seem to move it to the right ship.  Although it's showing up on my flagship in "Ship info", it's not letting me select it and it doesn't show up in any other screen:

![image](https://github.com/endless-sky/endless-sky/assets/406299/9230108c-acb7-448d-aed2-8a6d0b64590e)

Meh, I'll take a look at this at some other point.  I'm just getting very frustrated now, and that's not what I want from playing this.

**Comment by justinclift** [comment]
Awesome.  Yep, that worked:

![image](https://github.com/endless-sky/endless-sky/assets/406299/9edb7ffc-94f1-44b5-8ae4-26dd35f348e4)

Good workaround! :smile:

**Comment by movingpictures** [comment]
This is spectacular news, actually. It means the bug that allowed ships to
land on the gas giants (and to change flagships) has been fixed.

**Comment by ThrawnCA** [comment]
Good to know this is largely resolved, although one head-scratcher remains for me:

> Originally implemented because, seeing as the Drak want to preserve life, it seemed likely they'd figure out a way to prevent lower tier species from suiciding against them. So, the keep away field, which doubles as the get-away-from-me-before-my-turrets-kill-you field.

If they're so keen to preserve life, you'd think that they might, y'know, contact the player and warn you off, before escalating to _shooting you with an anti-matter cannon_. Do they want to kill you, in which case, it would make more sense to switch off the repelling field, or do they just want you to leave, in which case, they ought to say so?

**Comment by Zitchas** [comment]
In this particular situation, either:
a) they know who you are, and that you're working for the Remnant, or.
b) they don't know who you are (and/or don't care), and just assume that you're working for the Remnant.

In either case, they've already warned the Remnant extensively, and are simply treating you like they would any other member of the Remnant. 

As an aside, given how advanced the Archons are and the implications that they have vast capabilities beyond what we see; it wouldn't surprise if they *could* have some kind of faster and harder to dodge weapon; and that this AMC is probably a fairly primitive weapon as far as their arsenal is concerned. Chosen for the fact that it is big, flashy, and very impressive. Nothing like throwing a trident of death around to intimidate people. Yes, a strategy that can backfire, but most of the time it will have the desired effect.
